src/architect/cluster/cluster.utils.ts

```diff
@@ -81,4 +84,17 @@ export default class ClusterUtils {
     }
     return cluster;
   }
+
+  private static async getClientVersion() {
+    const { stdout } = await execa('kubectl', ['version', '--client', '--output', 'json']);
+    const { clientVersion } = JSON.parse(stdout);
+    return clientVersion;
+  }
+
+  public static async checkClientVersion(): Promise<void> {
+    const clientVersion = await this.getClientVersion();
+    if (Number(clientVersion.major) < MIN_CLUSTER_VERSION.major || (Number(clientVersion.major) === MIN_CLUSTER_VERSION.major && Number(clientVersion.minor) < 22)) {
```

Comment:
Use [semver](https://www.npmjs.com/package/semver) pkg to check version. It's already installed in the cli.

Reply:
I have updated to use semver to do version comparison.